feat: import search fields

mbritense opened the pull request:
Closes https://ritense.tpondemand.com/entity/87968-be-import-search-fields

mbritense committed:
search field importer

mbritense committed:
search field importer

mbritense committed:
test

theo-ritense commented on document/src/main/java/com/ritense/document/autoconfigure/SearchFieldAutoConfiguration.java:
```diff
@@ -51,6 +52,14 @@ public SearchFieldExporter searchFieldExporter(
+    @Bean
+    @ConditionalOnMissingBean(SearchFieldImporter.class)
+    public SearchFieldImporter searchFieldImporter(
+        final  SearchConfigurationDeploymentService searchConfigurationDeploymentService
```
```suggestion
        final SearchConfigurationDeploymentService searchConfigurationDeploymentService
```

theo-ritense commented on document/src/main/java/com/ritense/document/service/SearchConfigurationDeploymentService.java:
```diff
@@ -69,7 +70,7 @@ public void deployAll() {
     }
 
-    private void deploy(String documentDefinitionName, String searchConfigurationJson) throws IOException {
+    public void deploy(String documentDefinitionName, String searchConfigurationJson) throws IOException {
```
Since this is a new public method, can you write an (integration) test for it?

theo-ritense committed:
Added caselist column importer (#1100)

theo-ritense committed:
ImportService improvements (#1101)

- Added specific exceptions
- Added filter for Importers that depend on unprovided types

Klaas-Ritense committed:
#87966 Import process document association (#1093)

* #87966 Import process document association

* #87966 fx + test

* Apply suggestions from code review

Co-authored-by: theo-ritense <95682624+theo-ritense@users.noreply.github.com>

---------

Co-authored-by: theo-ritense <95682624+theo-ritense@users.noreply.github.com>

Klaas-Ritense committed:
#91264 case tab importer (#1102)

* #91264 case tab importer

marijnritense committed:
added importer for process definitions (#1103)

* added importer for process definitions

---------

Co-authored-by: theo-ritense <95682624+theo-ritense@users.noreply.github.com>

theo-ritense committed:
87971: Import case definition settings (#1105)

* 87971: Import case definition settings

* Update case/src/main/kotlin/com/ritense/case/service/CaseDefinitionSettingsImporter.kt

---------

Co-authored-by: marijnritense <94897451+marijnritense@users.noreply.github.com>

ivo-ritense committed:
Update document/src/main/java/com/ritense/document/autoconfigure/SearchFieldAutoConfiguration.java

Co-authored-by: theo-ritense <95682624+theo-ritense@users.noreply.github.com>

theo-ritense committed:
87969: Import decision definitions (#1106)

* 87969: Import decision definitions

* Update core/src/test/kotlin/com/ritense/valtimo/importer/CamundaDecisionDefinitionImporterTest.kt

Co-authored-by: Ivo Zaal <74657121+ivo-ritense@users.noreply.github.com>

---------

Co-authored-by: marijnritense <94897451+marijnritense@users.noreply.github.com>
Co-authored-by: Ivo Zaal <74657121+ivo-ritense@users.noreply.github.com>

theo-ritense committed:
87972 Import process links (#1104)

* Import process links

* inputstream fixes

---------

Co-authored-by: marijnritense <94897451+marijnritense@users.noreply.github.com>

mbritense committed:
pr fixes

mbritense committed:
Merge branch 'development/11.2.0' into feature/87968-import-search-fields
